release/xgboost_tests/workloads/ft_small_elastic.py

```diff
@@ -0,0 +1,50 @@
+"""Fault tolerance test (small cluster, elastic training)
+
+In this run, two training actors will die after some time. It is expected that
+in both cases xgboost_ray stops training, but continues right away with the
+remaining three actors. Shortly after, the actors will be restarted and
+re-integrated into the training loop. Training should finish with all four
+actors.
+
+Test owner: krfricke
+
+Acceptance criteria: Should run through and report final results. Intermediate
+output should show that training continues with fewer actors when an
+actor died. We may want to automate these checks.
+
+Notes: This test seems to be somewhat flaky. This might be due to
+race conditions in handling dead actors. This is likely a problem of
+the xgboost_ray implementation and not of this test.
+"""
+import ray
+from xgboost_ray import RayParams
+
+from _train import train_ray
+from ft_small_non_elastic import FailureState, FailureInjection
+
+if __name__ == "__main__":
+    ray.init(address="auto")
+
+    failure_state = FailureState.remote()
+
+    ray_params = RayParams(
+        elastic_training=True,
+        max_failed_actors=2,
+        max_actor_restarts=3,
+        num_actors=4,
+        cpus_per_actor=4,
+        gpus_per_actor=0)
+
+    train_ray(
+        path="/data/classification.parquet",
+        num_workers=4,
+        num_boost_rounds=100,
+        num_files=25,
+        regression=False,
+        use_gpu=False,
+        ray_params=ray_params,
+        xgboost_params=None,
+        callbacks=[
+            FailureInjection(state=failure_state, ranks=[3], iteration=14),
+            FailureInjection(state=failure_state, ranks=[0], iteration=34),
+        ])
```

Comment:
Do we need some assert statement here or some way for the release manager to make sure that the final number of actors is 4 (i.e. make sure that the 2 dead actors eventually come back alive)?

Reply:
Yes, we should probably introduce some more assertions here eventually. Right now I'm relying on visual inspection, but I'll see later if I find a quick way to confirm correct behavior (might adapt something from the unit tests).